github/misc_test.go

```diff
@@ -168,3 +168,39 @@ func TestRepositoriesService_ListServiceHooks(t *testing.T) {
 		t.Errorf("Repositories.ListServiceHooks returned %+v, want %+v", hooks, want)
 	}
 }
+
+func TestParseURL(t *testing.T) {
+	const (
+		owner, repo = "bazelbuild", "bazel"
+	)
+	tests := []struct {
+		url                 string
+		wantOwner, wantRepo string
+		wantErr             bool
+	}{
+		{url: "https://github.com/bazelbuild/bazel/tree/master", wantOwner: owner, wantRepo: repo},
+		{url: "http://github.com/bazelbuild/bazel/tree/master", wantOwner: owner, wantRepo: repo},
+		{url: "https://www.github.com/bazelbuild/bazel/tree/master", wantOwner: owner, wantRepo: repo},
+		{url: "https://github.com/bazelbuild/bazel/releases/tag/0.2.1", wantOwner: owner, wantRepo: repo},
+		{url: "https://github.com/bazelbuild/bazel/commit/19b...edc", wantOwner: owner, wantRepo: repo},
+		{url: "https://github.com/bazelbuild/bazel/archive/0.2.1.zip", wantOwner: owner, wantRepo: repo},
+		{url: "https://github.com/bazelbuild/bazel/", wantOwner: owner, wantRepo: repo},
+		{url: "https://github.com/bazelbuild/bazel", wantOwner: owner, wantRepo: repo},
+		{url: "https://github.com/bazelbuild/", wantOwner: owner},
+		{url: "https://github.com/bazelbuild", wantOwner: owner},
+		{url: "https://mygithub.com/bazelbuild", wantErr: true},
+		{url: "https://github.com/", wantErr: true},
+		{url: "https://github.com", wantErr: true},
+	}
+
+	for _, test := range tests {
+		gotOwner, gotRepo, err := ParseURL(test.url)
+		if err != nil && !test.wantErr {
```

Comment:
There are many ways to fix it, pick your favorite. Some examples:

- https://github.com/shurcooL/gostatus/blob/f8723cde71fe2c54eb101408c/status/url_test.go#L60-L66
- https://github.com/sourcegraph/sourcegraph/blob/47f606ac15d64/pkg/routevar/user_test.go#L135-L145
- https://github.com/sourcegraph/sourcegraph/blob/47f606ac15d64/app/internal/returnto/returnto_test.go#L32-L39